src/components/Search/index.tsx

```diff
@@ -13,127 +13,74 @@ import { useHistory } from 'react-router'
-        break
-
-      case SearchResultType.UDT:
-        history.push(data.attributes.udtType === 'omiga_inscription' ? `/inscription/${query}` : `/sudt/${query}`)
-        break
-
-      default:
-        break
-    }
-  } catch (error) {
-    setInputLoading(false)
-
-    if (
-      isAxiosError(error) &&
-      error.response?.data &&
-      error.response.status === 404 &&
-      (error.response.data as Response.Error[]).find(
-        (errorData: Response.Error) => errorData.code === HttpErrorCode.NOT_FOUND_ADDRESS,
-      )
-    ) {
-      onClear()
-      history.push(`/address/${query}`)
-    } else {
-      history.push(`/search/fail?q=${query}`)
-    }
-  }
-}
+// Currently, the API returns all search results, which could be extremely large in quantity.
+// Since the rendering component does not implement virtual scrolling, this leads to a significant decrease in page performance.
+// Therefore, here we are implementing a frontend-level limitation on the number of displayed results.
+const DISPLAY_COUNT = 10
 
 const Search: FC<{
   content?: string
   hasButton?: boolean
   onEditEnd?: () => void
 }> = memo(({ content, hasButton, onEditEnd: handleEditEnd }) => {
-  // Currently, the API returns all search results, which could be extremely large in quantity.
-  // Since the rendering component does not implement virtual scrolling, this leads to a significant decrease in page performance.
-  // Therefore, here we are implementing a frontend-level limitation on the number of displayed results.
-  const DISPLAY_COUNT = 10
-  const isMobile = useIsMobile()
+  const history = useHistory()
   const queryClient = useQueryClient()
   const { t } = useTranslation()
-  const [searchByType, setSearchByType] = useSearchType()
-  const [isSearchByName, setIsSearchByName] = useState(searchByType === 'name')
-  const history = useHistory()
+  const isMobile = useIsMobile()
+  const [searchType, setSearchType] = useSearchType()
+  const isSearchByName = searchType === 'name'
+
   const [keyword, setKeyword] = useState(content || '')
-  const [editEnded, setEditEnded] = useState(false)
+  const [editEnded, _setEditEnded] = useState(false)
   const searchValue = keyword.trim()
-  const [inputLoading, setInputLoading] = useState(false)
-
-  const toggleSearchType = () => {
-    const newIsSearchByNames = !isSearchByName
-    const searchTypePersistValue = newIsSearchByNames ? 'name' : 'id'
-    setSearchByType(searchTypePersistValue)
-    setIsSearchByName(newIsSearchByNames)
-    queryClient.resetQueries(['searchByName'])
-  }
+
+  const setEditEnded = useCallback(
+    (value: boolean) => {
+      _setEditEnded(value)
+      if (value) handleEditEnd?.()
+    },
+    [handleEditEnd],
+  )
+
+  const {
+    refetch: refetchSearchById,
+    data: urlByIdSearch,
+    isFetching: isFetchingById,
+  } = useQuery(['searchById', searchValue], () => getURLByIdSearch(addPrefixForHash(searchValue)), {
+    enabled: false,
```

Comment:
Why set this query disabled

Reply:
Because the user's last input must be used for the query here, there is no need to query in advance. If you want to query in advance, it will greatly increase the complexity of the code.

Reply:
Got it